### ISSUE TITLE
buffer overflow happens if interface name too long

### ISSUE THREAD

**Issue by leona-ya** [ppp-project/ppp#488]
Scenario: A user wants to use pppd with pppoe on an interface with a linux automated name. Here the interface is named `enp195s0f3u1u4` (note it has 14 characters and the linux kernel limit [is 16 characters](https://elixir.bootlin.com/linux/v6.8/source/include/uapi/linux/if.h#L33).

PPP config (/etc/ppp/peers/one):
```
plugin pppoe.so

# interface name
enp195s0f3u1u4

name [redacted]

debug

persist
maxfail 0
holdoff 5

noipdefault
hide-password
defaultroute
noauth
lcp-echo-interval 20
lcp-echo-failure 3
mtu 1492
ifname ppp-wan
+ipv6
```

and it fails with
```
❯ sudo pppd call one nodetach
Plugin pppoe.so loaded.
PPPoE plugin from pppd 2.5.0
*** buffer overflow detected ***: terminated
Fatal signal 6
```
  

It just works when I rename the interface to `wan`.

**Comment by paulusmack** [comment]
What C compiler are you using and do you have any additional compiler options enabled? Have you set _FORTIFY_SOURCE for example, and if so to what?

I tried to reproduce this problem locally without success.

**Comment by Neustradamus** [comment]
@leona-ya: Have you seen the @paulusmack questions?

**Comment by paulusmack** [comment]
I tested the current source with -D_FORTIFY_SOURCE=3 and the same long interface name as in the original report, and didn't see any failure. I think this may have been fixed by either f19193e0 or 52a531f7.

@leona-ya can you test with the current git source code?

**Comment by leona-ya** [comment]
With current git it works. Wonderful, thanks.
